server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -503,11 +502,11 @@ private boolean isReadyForUpload() {
             if (indexShard.getEngineOrNull() != null) {
                 sb.append(" engineType=").append(indexShard.getEngine().getClass().getSimpleName());
             }
-            if (isLocalOrSnapshotRecovery() == false) {
+            if (indexShard.recoveryState() != null) {
                 sb.append(" recoverySourceType=").append(indexShard.recoveryState().getRecoverySource().getType());
                 sb.append(" primary=").append(indexShard.shardRouting.primary());
             }
-            logger.trace(sb.toString());
+            logger.info(sb.toString());
```

Comment:
Why info?

Reply:
This is helpful in debugging cases where the refresh listener's refresh method is invoked but it is not ready for upload due to any reason.

Reply:
There will be very few instances when this will be logged. for eg - create index, failover, relocation. This will not bombard the logs.